### PR TITLE
Pull-through using insecure transport based on tag import policy

### PR DESCRIPTION
**Pull request by miminar**
Insecure istag now allows for insecure transport.

Resolves: [bz#1421954](https://bugzilla.redhat.com/show_bug.cgi?id=1421954)

### CODE REVIEW

**Comment by legionus** [comment]
@miminar I broke your PR. Sorry.

**Comment by miminar** [comment]
@legionus :smile: no problem. I did the same to you. Fixed now.

File: pkg/dockerregistry/server/repositorymiddleware.go [hunk 1]
```diff
@@ -143,6 +147,8 @@ type repository struct {
 	acceptschema2 bool
 	// blobrepositorycachettl is an eviction timeout for <blob belongs to repository> entries of cachedLayers
 	blobrepositorycachettl time.Duration
+	// cachedImageStream stays cached for the entire time of handling signle repository-scoped request.
+	cachedImageStream *imageapi.ImageStream
```
Comment (legionus) [inline pkg/dockerregistry/server/repositorymiddleware.go]: Can we use LRU globally and cache imagestream for few seconds (1-3) ?

**Comment by miminar** [comment]
[test]

**Comment by miminar** [comment]
There's no need for setting insecure annotation on image stream during image import. The insecure tag policy gets propagated to tag spec into the image stream. The registry now honors this tag-based policy and uses insecure transport if the image has been imported as insecure.

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**